extensions/3rdparty/LyricWiki/Special_LinksToRedirects.php

```diff
@@ -139,107 +136,105 @@ function execute( $par ){
-							$pageId = mysql_result($result, $cnt, "from_id");
-							$target = mysql_result($result, $cnt, "links_to");
-							$allListings[] = array($pageId, $target);
-
-							// Somehow, a blank from_id can slip into the results (which messes up the query).
-							if($pageId != ""){
-								$ids[] = $pageId; // will uniquify later
-							}
-						}
+				
+				$dbr = wfGetDB( DB_SLAVE );
+				$res = $dbr->query($queryString);
+				$totalResults = 0;
+				while ($row = $dbr->fetchObject($res)) {
+					$totalResults++;
+					$pageId = $row->from_id;
+					$target = $row->links_to;
+					$allListings[] = array($pageId, $target);
+
+					// Somehow, a blank from_id can slip into the results (which messes up the query).
+					if($pageId != ""){
+						$ids[] = $pageId; // will uniquify later
+					}
+				}
 
-						// Find the page titles of all of the source pages by id (uniquify the id array).
-						GLOBAL $wgSitename;
-						$nsMapping = array(
-										1 => "Talk",
-										2 => "User",
-										3 => "User_talk",
-										4 => $wgSitename,
-										5 => $wgSitename."_talk",
-										6 => ":File",
-										7 => "File_talk",
-										8 => "MediaWiki",
-										9 => "MediaWiki_talk",
-										10 => ":Template",
-										11 => "Template_talk",
-										12 => "Help",
-										13 => "Help_talk",
-										14 => ":Category",
-										15 => "Category_talk"
-									);
-						$idToTitle = array();
-						$ids = array_unique($ids);
-						$queryString = "SELECT page_id, page_namespace, page_title FROM $TABLE_PREFIX"."page WHERE page_id IN (".implode(",", $ids).")";
-						if($result = mysql_query($queryString,$db)){
-							if(($numRows = mysql_num_rows($result)) && ($numRows > 0)){
-								for($cnt=0; $cnt<$numRows; $cnt++){
-									$id = mysql_result($result, $cnt, "page_id");
-									$ns = mysql_result($result, $cnt, "page_namespace");
-									$title = mysql_result($result, $cnt, "page_title");
-
-									if(isset($nsMapping[$ns])){
-										$title = $nsMapping[$ns].":$title";
-									}
-
-									$idToTitle[$id] = $title;
-								}
-							}
+				if($totalResults == 0){
+					print "<em>No more links to redirects found. <strong>(Yay!)</strong></em>\n";
+				} else {
+					// Find the page titles of all of the source pages by id (uniquify the id array).
+					GLOBAL $wgSitename;
+					$nsMapping = array(
+						1 => "Talk",
+						2 => "User",
+						3 => "User_talk",
+						4 => $wgSitename,
+						5 => $wgSitename."_talk",
+						6 => ":File",
+						7 => "File_talk",
+						8 => "MediaWiki",
+						9 => "MediaWiki_talk",
+						10 => ":Template",
+						11 => "Template_talk",
+						12 => "Help",
+						13 => "Help_talk",
+						14 => ":Category",
+						15 => "Category_talk"
+					);
+					$idToTitle = array();
+					$ids = array_unique($ids);
+					$queryString = "SELECT page_id, page_namespace, page_title FROM $TABLE_PREFIX"."page WHERE page_id IN (".implode(",", $ids).")";
```

Comment:
Let's be consistent with the rest of the change and use `Database::select` helper instead of providing a raw SQL query.